core/runtime/bindings.rs

```diff
@@ -368,6 +369,31 @@ pub extern "C" fn host_initialize_import_meta_object_callback(
   let resolve_key =
     v8::String::new_external_onebyte_static(scope, b"resolve").unwrap();
   meta.set(scope, resolve_key.into(), val.into());
+
+  // For `file:` URL we provide additional `filename` and `dirname` values
+  let Ok(name_url) = Url::parse(&name) else {
+    return;
+  };
+
+  if name_url.scheme() == "file" {
+    let file_path = name_url.to_file_path().unwrap();
+
+    let filename_key =
+      v8::String::new_external_onebyte_static(scope, b"filename").unwrap();
+    let dirname_key =
+      v8::String::new_external_onebyte_static(scope, b"dirname").unwrap();
+    // Use display() here so that Rust takes care of proper forward/backward slash
+    // formatting depending on the Os.
+    let filename_val =
+      v8::String::new(scope, &file_path.display().to_string()).unwrap();
+
+    let dir_path = file_path.parent().unwrap();
```

Comment:
This is probably wrong for URL like `file:///some_file.js` - ie. it works fine on Unix, but on Windows I'm not sure what we should do here 🤔

Reply:
A module wouldn't have that kind of URL on windows so that's ok.

Reply:
This is probably only a problem for testing code - should I just default to `/` here?

Reply:
In the test code we should do `file:///C:/some_file.js` on Windows.

Reply:
That would require changing pretty much all tests - can I just default to `/` in such case?